config/always-python.m4

```diff
@@ -47,41 +58,42 @@ AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_PYTHON], [
 
 	AS_CASE([$with_python],
 		[check],
-		[AS_IF([test -x /usr/bin/python3],
+		[AS_IF([test -x "$(which python3)"],
```

Comment:
What is the point of using `which` + `test` here, rather than just `which`? In other words, does this work just as well:
`AS_IF([which python3 > /dev/null],`

Reply:
Happy to make that change. Works for me on FreeBSD.

Reply:
The "which" command is nonstandard and various systems have produced wildly different implementations, the worst of which will print a csh interpreter alias instead of a command binary *even from a non-interactive script*. https://unix.stackexchange.com/a/86029/264819

Please don't use `which` for any purposes -- treat it like the cardinal sin of http://mywiki.wooledge.org/ParsingLs :D

Instead use the POSIX-mandated `command -v`, which must be present according to POSIX 2008, a decade-old standard.

Reply:
For bonus points tear out the whole thing and just use the reliable autoconf standard, which is `AC_CHECK_PROG`.

Reply:
Thanks for the inspiration, `AC_CHECK_PROGS` does exactly what is needed here.

Reply:
(Note that this autotools macro is back-compatible with systems that have never heard of POSIX 2008 at all -- it will expand to a manual `IFS=:` followed by looping over $PATH and checking whether each candidate exists via `test -f && test -x`.)